data/blog/2025/2025-06-02-code-lifecycle.md

```diff
@@ -0,0 +1,109 @@
+
+No matter how well the architecture is defined at first, a constant development process may lead to unwanted shortcuts that slowly build on technical debt.
+Eventually, implementing a new feature means rewriting too much code or adding potential instability.
+
+### Keep the buzz going or people forget about you
+
+The original expression is 'strike while the iron is hot', but in our case, the 'iron' (software) remains hot only when it is struck (maintained).
+
+Since there is often indirect competition between frameworks and packages, some will continually gain popularity and followers, while others naturally fade into anonymity.
+The **go-to option** will too often be the more **popular package** with an **active community**, a pretty **marketing campaign**, while the lesser-popular one stagnates on feature implementation.
+Marketing takes many forms:
+
+- The professional (and expensive) approach.
+- Keeping the community engaged with:
+    - Regular updates,
+    - New features,
+    - Discussions,
+    - Articles and the like.
+
+## Reasons to keep the code up-to-date
+
+Abandoning code is easy, but there are reasons to keep code alive and well-maintained.
+
+Widespread usage often guarantees a large community supporting a package with contribution in both code and donations.
+Donations from the open source may not help the developer retire any time soon, but it can help in the short term to keep the package running well.
+Obtaining a more reliable source of donations from the private or government sector does certainly go a long way toward keeping devs focused.
+
+## How Laminas handles the code lifecycle
+
+In recent months, the Laminas Technical Steering Committee has been **reviewing** their long list of **packages** in order to decide what to abandon and what to keep active.
+During the [TSC meeting on 2024-11-04](https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2024-11-04-TSC-Minutes.md), several packages in security-mode were discussed and the ultimate decision was to mark most of them discontinued.
+
+A similar decision was made for [Laminas MVC](https://github.com/laminas/laminas-mvc), though the decision is much more impactful.
+There are many projects that still use Laminas MVC and are actively supported by teams that still receive feature requests.
+Most of the recent development effort from the Laminas members has shifted to [Laminas ServiceManager 4](https://github.com/laminas/laminas-servicemanager) (which implements [PSR-11 Container version 2](https://github.com/php-fig/container)) and the middleware architecture.
+For this reason, it's unreasonable to keep Laminas MVC up to date which requires a great deal of attention from TSC members who simply have no more time for it.
+**Archiving Laminas MVC** will be a **slow process** to make sure that its users have plenty of time to move to an alternative, like the recommended middleware architecture.
```

Comment:
An official announcement should be made for this, otherwise it will look like we are hiding this important information under a different headline.

Reply:
you're right. i created an article for this announcement too https://github.com/laminas/getlaminas.org/pull/268